console/src/components/Services/Data/Schema/ConnectDatabase.tsx

```diff
@@ -0,0 +1,769 @@
+  }, [isEditState, currentSourceInfo]);
+
+  const crumbs = [
+    {
+      title: 'Data',
+      url: `/data/${props.currentDataSource}/schema/${props.currentSchema}`,
+    },
+    {
+      title: 'Manage Databases',
+      url: '/data/manage',
+    },
+    {
+      title: 'Connect Database',
+      url: '#',
+    },
+  ];
+
+  const onChangeConnectionType = (e: ChangeEvent<HTMLInputElement>) => {
+    changeConnectionType(e.target.value);
+  };
+
+  const { dispatch } = props;
+
+  const resetState = () => {
+    connectDBDispatch({
+      type: RESET_INPUT_STATE,
+      data: null,
+    });
+  };
+
+  const onClickConnectDatabase = () => {
+    if (!connectDBInputState.displayName.trim()) {
+      dispatch(
+        showErrorNotification(
+          'Display Name is a mandatory field',
+          'Please enter a valid display name.'
+        )
+      );
+      return;
+    }
+
+    if (isEditState) {
+      dispatch(
+        editDataSource(
+          oldName,
+          {
+            driver: connectDBInputState.dbType,
+            payload: {
+              name: connectDBInputState.displayName.trim(),
+              dbUrl: connectDBInputState.databaseURLState.dbURL,
+              connection_pool_settings: connectDBInputState.connectionSettings,
+            },
+          },
+          () => {
+            resetState();
+            dispatch(_push('/data/manage'));
+          }
+        )
+      );
+      return;
+    }
+
+    if (connectionType === connectionTypes.DATABASE_URL) {
+      if (!connectDBInputState.databaseURLState.dbURL.trim()) {
+        dispatch(
+          showErrorNotification(
+            'Database URL is a mandatory field',
+            'Please enter a valid database URL'
+          )
+        );
+        return;
+      }
+
+      dispatch(
+        addDataSource(
+          {
+            driver: connectDBInputState.dbType,
+            payload: {
+              name: connectDBInputState.displayName.trim(),
+              dbUrl: connectDBInputState.databaseURLState.dbURL,
+              connection_pool_settings: connectDBInputState.connectionSettings,
+            },
+          },
+          () => resetState()
+        )
+      );
+      return;
+    }
+
+    if (connectionType === connectionTypes.ENV_VAR) {
+      if (!connectDBInputState.databaseURLState.dbURL.trim()) {
+        dispatch(
+          showErrorNotification(
+            'Environment Variable is a mandatory field',
+            'Please enter the name of a valid environment variable'
+          )
+        );
+        return;
+      }
+
+      dispatch(
+        addDataSource(
+          {
+            driver: connectDBInputState.dbType,
+            payload: {
+              name: connectDBInputState.displayName.trim(),
+              dbUrl: connectDBInputState.envVarURLState.envVarURL,
+              connection_pool_settings: connectDBInputState.connectionSettings,
+            },
+          },
+          () => resetState()
+        )
+      );
```

Comment:
This part is largely similar for most use cases while adding/editing data sources, we can wrap this with a helper